### PR TITLE
test: atomic operation : add test case

### PR DESCRIPTION
**Pull request by yingmingx**
Add tests with negative parameters to supplement black box tests.

Signed-off-by: Ying ming <mingx.ying@intel.com>

### CODE REVIEW

File: tests/kernel/common/src/atomic.c [hunk 1]
```diff
@@ -114,12 +114,20 @@ void test_atomic(void)
...
+	value = -4;
+	zassert_true((atomic_add(&target, value) == 3), "atomic_add");
+	zassert_true((target == -1), "atomic_add");
 
 	/* atomic_sub() */
 	target = 10;
 	value = 2;
 	zassert_true((atomic_sub(&target, value) == 10), "atomic_sub");
 	zassert_true((target == 8), "atomic_sub");
+	/* Test function parameters can be negative */
```
Comment (LeiW000) [inline tests/kernel/common/src/atomic.c:127]: Same as line 117.

**Review by LeiW000 — CHANGES_REQUESTED** [review]

**Review by LeiW000 — APPROVED** [review]

**Review by enjiamai — CHANGES_REQUESTED** [review]
Soft reminder, I saw you make code change base on Steven's comment, but I still saw that comment existing (at least it should show "outdated"). Better to have it disappeared then inform Steven to review it again.

**Comment by yingmingx** [comment]
> Soft reminder, I saw you make code change base on Steven's comment, but I still saw that comment existing (at least it should show "outdated"). Better to have it disappeared then inform Steven to review it again.

Hi, enjia. Thank you for you careful review. I have found the reason why this comment is existing because I add the new code below the line Steven have comment on. Now I change the notes which the line Steven comment on it. It's Ok now.

**Review by enjiamai — APPROVED** [review]

**Review by LeiW000 — APPROVED** [review]

**Comment by LeiW000** [comment]
@andyross , could you help to merge it?

**Comment by LeiW000** [comment]
@nashif @andrewboie , could you help to take look at it and merge?

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**